### PR TITLE
Proposal to add Keyboarding APIs

### PR DESCRIPTION
**Pull request by harinikmsft**
React Native components and apps need the ability to handle key strokes and implement custom logic when end users are interacting with a React Native app using a Keyboard. This proposal outlines the basic props and events needed to enable this scenario.

### CODE REVIEW

File: proposals/0000-keyboard-apis.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+|:---:|:----:|:----:|
+| keyDownEvents | IHandledKeyboardEvents[] | Specifies the key(s) that are handled in the JS layer by the onKeyDown/onKeyDownCapture events |
+| keyUpEvents | IHandledKeyboardEvents[] | Specifies the key(s) that are handled in the JS layer by the onKeyUp/onKeyUpCapture events |
+
+Where `IHandledKeyboardEvents` is a new type which takes a string parameter named `key` to declare the key strokes that are of interest to the JS layer.
+
+When the `onKeyXX` events are handled by the app code, the corresponding native component will have KeyXX events marked as *handled* for the declared key strokes.
+
+### TBD
+
```
Comment (harinikmsft) [inline proposals/0000-keyboard-apis.md:122]: @kmelmon , @StephenLPeters  - can you add details for this section?

File: proposals/0000-keyboard-apis.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+## Summary
+
+Components need the ability to handle key strokes and implement custom logic when interacting with a React Native app using Keyboard. This proposal outlines the basic props and events needed to enable this scenario.
+
+## Motivation
+
+There are several usecases where end users may interact with `react-native` applications using keyboard. These include:
+- iPads and Android tablet devices with attached keyboards or bluetooth keyboards
+- `react-native-windows` apps that will be deployed on PCs and laptop computers
```
Comment (stecrain) [inline proposals/0000-keyboard-apis.md:17]: Xbox too! :D unless there are different apis planned for keyboard vs controller?

File: proposals/0000-keyboard-apis.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+| onKeyDownCapture | IKeyboardEvent | void | Occurs when the `onKeyDown` event is being routed. `onKeyDown` is the corresponding bubbling event. On Windows, this corresponds to [PreviewKeyDown](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.uielement.previewkeydown)|
+| onKeyUp | IKeyboardEvent | void | Occurs when a keyboard key is released when a component has focus. On Windows, this corresponds to [KeyUp](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.uielement.keyup)|
+| onKeyUpCapture | IKeyboardEvent | void | Occurs when the `onKeyUp` event is being routed. `onKeyUp` is the corresponding bubbling event. On Windows, this corresponds to [PreviewKeyUp](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.uielement.previewkeyup)|
+
+Where `IKeyboardEvent` will be a new event type added to `ReactNative.NativeSyntheticEvents` of type `INativeKeyboardEvent`. `INativeKeyboardEvent` is a new interface and will expose the following properties:
```
Comment (stecrain) [inline proposals/0000-keyboard-apis.md:83]: Does this describe the properties that will be passed to the javascript layer or is this specifically for the native interface?

If this implies the javascript event object as well it seems like there are some things missing from the react synthetic event, such as charCode

File: proposals/0000-keyboard-apis.md (hunk 2 above)
Comment (harinikmsft) [reply proposals/0000-keyboard-apis.md:17]: There will be some more APIs for XYFocus that are specific to Xbox. But, you are correct - these APIs will also benefit gamepad invocation scenarios. I will add that in.

File: proposals/0000-keyboard-apis.md (hunk 3 above)
Comment (harinikmsft) [reply proposals/0000-keyboard-apis.md:83]: This is for the JS layer. The charCode might be interesting if we were to add onKeyPress in addition to KeyDown/KeyUP. For now, we are not adding KeyPress - we can discuss this during the review to see if there are scenarios impacted right away that need it.

**Pull request closed without merging**